crates/bevy_ecs/src/system/system.rs

```diff
@@ -37,6 +39,24 @@ pub trait System: Send + Sync + 'static {
     }
     fn apply_buffers(&mut self, world: &mut World);
     fn initialize(&mut self, _world: &mut World);
+    fn check_system_counter(&mut self, global_system_counter: u32);
 }
 
 pub type BoxedSystem<In = (), Out = ()> = Box<dyn System<In = In, Out = Out>>;
+
+pub(crate) fn check_system_counter_impl(
```

Comment:
I'm not quite able to follow the logical flow between the many files: do we end up running this function on systems that do not contain any change-detecting parameters? If so, can we short-circuit?

Reply:
Currently yes. Short-circuiting is not an important optimization since the check is only made occasionally (after 500 million systems run), and number of systems is small compared to the number of components. It should be done at some point though, if only to avoid triggering the warning on unrelated systems, but I need to think about how to do it.

Reply:
> if only to avoid triggering the warning on unrelated systems

Yeah, this is my thinking. That would be a *very* confusing warning to get, but the performance difference is totally negligible.